src/linux/processor.rs

```diff
@@ -228,27 +228,36 @@ pub fn get_raw_times(p: &Processor) -> (u64, u64) {
 }
 
 pub fn get_cpu_frequency() -> u64 {
-    // /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq
     let mut s = String::new();
+    if File::open("/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq")
+        .and_then(|mut f| f.read_to_string(&mut s))
+        .is_ok()
+    {
+        let freq_option = s.trim().split('\n').next();
+        if let Some(freq_string) = freq_option {
+            if let Ok(freq) = freq_string.parse::<u64>() {
+                return freq / 1000;
+            }
+        }
+    }
+    s.clear();
     if File::open("/proc/cpuinfo")
         .and_then(|mut f| f.read_to_string(&mut s))
-        .is_err()
+        .is_ok()
```

Comment:
In this case however you can check `is_err` and return 0 in case it's an error so we can get one indent level haha.